implementations/rust/ockam/ockam_entity/src/channel.rs

```diff
@@ -11,32 +11,51 @@ pub(crate) use listener::*;
 mod messages;
 pub(crate) use messages::*;
 
-impl<V: ProfileVault> ProfileImpl<V> {
+#[allow(dead_code)]
+struct SecureChannel {}
+
+impl SecureChannel {
     /// Create mutually authenticated secure channel
-    pub async fn create_secure_channel<A: Into<Route>>(
-        &mut self,
+    #[allow(dead_code)]
+    pub async fn create_secure_channel<R, V, P>(
         ctx: &Context,
-        route: A,
-    ) -> Result<Address> {
-        Initiator::create(ctx, route, self).await
+        profile: &mut P,
+        route: R,
+    ) -> Result<Address>
+    where
+        R: Into<Route>,
+        V: ProfileVault,
+        P: ProfileTrait<V>,
+    {
+        Initiator::create(ctx, route, profile).await
     }
 
     /// Create mutually authenticated secure channel listener
-    pub async fn create_secure_channel_listener<A: Into<Address>>(
-        &mut self,
+    #[allow(dead_code)]
```

Comment:
It looks like this function doesn't belong to crate's public API, otherwise it wouldn't be dead_code

Reply:
Let's pair after demos tomorrow, I'll go through what I did and see where I got stuck